components/script/dom/bindings/str.rs

```diff
@@ -306,11 +306,19 @@ impl DOMString {
     /// A valid number is the same as what rust considers to be valid,
     /// except for +1., NaN, and Infinity.
     /// https://html.spec.whatwg.org/multipage/#valid-floating-point-number
-    pub fn is_valid_number_string(&self) -> bool {
-        let input = &self.0;
-        input.parse::<f64>().ok().map_or(false, |val| {
-            !(val.is_infinite() || val.is_nan() || input.ends_with(".") || input.starts_with("+"))
-        })
+    pub fn is_valid_float_number_string(&self) -> bool {
+        parse_float_number(&self.0).is_ok()
+    }
+
+    /// https://html.spec.whatwg.org/multipage/#best-representation-of-the-number-as-a-floating-point-number
+    pub fn best_representation_of_the_float_number(&mut self) {
```

Comment:
IMO, maybe naming the function with a `set` prefix might be better to let other developers to know this function will `set` value.

Reply:
good point!